VultisigApp/VultisigApp/Model/Coin.swift

```diff
@@ -172,11 +173,11 @@ class Coin: ObservableObject, Codable, Hashable {
     
     func fiat(value: BigInt) -> Decimal {
         let decimal = decimal(for: value)
-        return decimal * Decimal(priceRate)
+        return RateProvider.shared.fiatBalance(value: decimal, coin: self)
```

Comment:
I would declare the RateProvider as a var, so we don't have to change everywhere if we need to change it.